py-polars/polars/utils/convert.py

```diff
@@ -97,52 +95,49 @@ def _negate_duration(duration: str) -> str:
     return f"-{duration}"
 
 
+def _time_to_pl_time(t: time) -> int:
+    t = t.replace(tzinfo=timezone.utc)
+    return (
+        t.hour * 3_600 + t.minute * 60 + t.second
+    ) * 1_000_000_000 + t.microsecond * 1_000
+
+
+def _date_to_pl_date(d: date) -> int:
+    dt = datetime.combine(d, datetime.min.time()).replace(tzinfo=timezone.utc)
+    return int(dt.timestamp()) // SECONDS_PER_DAY
+
+
 def _datetime_to_pl_timestamp(dt: datetime, time_unit: TimeUnit | None) -> int:
     """Convert a python datetime to a timestamp in given time unit."""
     if dt.tzinfo is None:
         # Make sure to use UTC rather than system time zone.
         dt = dt.replace(tzinfo=timezone.utc)
+    micro_seconds = dt.microsecond
+    seconds = _timestamp_in_seconds(dt)
     if time_unit == "ns":
-        micros = dt.microsecond
-        return 1_000 * (_timestamp_in_seconds(dt) * 1_000_000 + micros)
+        return 1_000 * (seconds * 1_000_000 + micro_seconds)
     elif time_unit == "us" or time_unit is None:
-        micros = dt.microsecond
-        return _timestamp_in_seconds(dt) * 1_000_000 + micros
+        return seconds * 1_000_000 + micro_seconds
     elif time_unit == "ms":
-        millis = dt.microsecond // 1000
-        return _timestamp_in_seconds(dt) * 1_000 + millis
-    else:
-        raise ValueError(
-            f"`time_unit` must be one of {{'ns', 'us', 'ms'}}, got {time_unit!r}"
-        )
-
-
-def _time_to_pl_time(t: time) -> int:
-    t = t.replace(tzinfo=timezone.utc)
-    return int((t.hour * 3_600 + t.minute * 60 + t.second) * 1e9 + t.microsecond * 1e3)
-
-
-def _date_to_pl_date(d: date) -> int:
-    dt = datetime.combine(d, datetime.min.time()).replace(tzinfo=timezone.utc)
-    return int(dt.timestamp()) // (3600 * 24)
+        return seconds * 1_000 + micro_seconds // 1_000
+    raise ValueError(
+        f"`time_unit` must be one of {{'ms', 'us', 'ns'}}, got {time_unit!r}"
+    )
```

Comment:
Same here, not sure this is an improvement. Though extracting the `seconds` calculation is a good move.

Reply:
This specific block handles `timestamp`, and it's indeed not an improvement. It does, however, make the structure similar to the function that handles `timedelta`